### ISSUE TITLE
[arm32/Ubuntu] Release build failures

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#16996]
Failures in arm32/Ubuntu release Pri-1 test run.

https://ci.dot.net/job/dotnet_coreclr/job/dev_unix_test_workflow/job/arm_cross_release_ubuntu_tst_prtest/1/consoleText

```
FAILED   - JIT/HardwareIntrinsics/X86/Avx2/Avx2_ro/Avx2_ro.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_release_ubuntu_tst_prtest/bin/tests/Linux.arm.Release/Tests/Core_Root/corerun Avx2_ro.exe
               Running Add.Byte test...
               Running Add.Int16 test...
               Running Add.Int32 test...
               Running Add.Int64 test...
               Running Add.SByte test...
               Running Add.UInt16 test...
               Running Add.UInt32 test...
               Running Add.UInt64 test...
               Running And.Byte test...
               Running And.Int16 test...
               Running And.Int32 test...
               Running And.Int64 test...
               Running And.SByte test...
               Running And.UInt16 test...
               Running And.UInt32 test...
               Running And.UInt64 test...
               Running AndNot.Byte test...
               Running AndNot.Int16 test...
               Running AndNot.Int32 test...
               Running AndNot.Int64 test...
               Running AndNot.SByte test...
               Running AndNot.UInt16 test...
               Running AndNot.UInt32 test...
               Running AndNot.UInt64 test...
               Running Average.Byte test...
               Running Average.UInt16 test...
               Running BlendVariable.Byte test...
               Running BlendVariable.SByte test...
               Running BroadcastScalarToVector128.Byte test...
               Running BroadcastScalarToVector128.SByte test...
               Running BroadcastScalarToVector128.Int16 test...
               Running BroadcastScalarToVector128.UInt16 test...
               Running BroadcastScalarToVector128.Int32 test...
               Running BroadcastScalarToVector128.UInt32 test...
               Running BroadcastScalarToVector128.Int64 test...
               Running BroadcastScalarToVector128.UInt64 test...
               Running BroadcastScalarToVector128.Single test...
               Running BroadcastScalarToVector128.Double test...
               Running BroadcastScalarToVector256.Byte test...
               Running BroadcastScalarToVector256.SByte test...
               Running BroadcastScalarToVector256.Int16 test...
               Running BroadcastScalarToVector256.UInt16 test...
               Running BroadcastScalarToVector256.Int32 test...
               Running BroadcastScalarToVector256.UInt32 test...
               Running BroadcastScalarToVector256.Int64 test...
               Running BroadcastScalarToVector256.UInt64 test...
               Running BroadcastScalarToVector256.Single test...
               Running BroadcastScalarToVector256.Double test...
               Running CompareEqual.Byte test...
               Running CompareEqual.Int16 test...
               Running CompareEqual.Int32 test...
               Running CompareEqual.Int64 test...
               Running CompareEqual.SByte test...
               Running CompareEqual.UInt16 test...
               Running CompareEqual.UInt32 test...
               Running CompareEqual.UInt64 test...
               Running CompareGreaterThan.Int16 test...
               Running CompareGreaterThan.Int32 test...
               Running CompareGreaterThan.Int64 test...
               Running CompareGreaterThan.SByte test...
               Running Or.Byte test...
               Running Or.Int16 test...
               Running Or.Int32 test...
               Running Or.Int64 test...
               Running Or.SByte test...
               Running Or.UInt16 test...
               Running Or.UInt32 test...
               Running Or.UInt64 test...
               Running ShiftLeftLogical.Int16.1 test...
               ./Avx2_ro.sh: line 243: 21310 Segmentation fault      $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
               Expected: 100
               Actual: 139
               END EXECUTION - FAILED
FAILED   - JIT/Methodical/MDArray/DataTypes/float_cs_r/float_cs_r.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_release_ubuntu_tst_prtest/bin/tests/Linux.arm.Release/Tests/Core_Root/corerun float_cs_r.exe
               ./float_cs_r.sh: line 243: 24559 Segmentation fault      $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
               Expected: 100
               Actual: 139
               END EXECUTION - FAILED
FAILED   - JIT/Regression/VS-ia64-JIT/V1.2-M02/b26496/b26496/b26496.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_release_ubuntu_tst_prtest/bin/tests/Linux.arm.Release/Tests/Core_Root/corerun b26496.exe
               Testing AA::Static1
               ./b26496.sh: line 243: 26616 Segmentation fault      $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
               Expected: 100
               Actual: 139
               END EXECUTION - FAILED
FAILED   - reflection/Tier1Collectible/Tier1Collectible/Tier1Collectible.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/dev_unix_test_workflow/arm_cross_release_ubuntu_tst_prtest/bin/tests/Linux.arm.Release/Tests/Core_Root/corerun Tier1Collectible.exe
               coreclr_initialize failed - status: 0x80131534
               Expected: 100
               Actual: 255
               END EXECUTION - FAILED

=======================
     Test Results
=======================
# CoreCLR Bin Dir  : 
# Tests Discovered : 5
# Passed           : 1
# Failed           : 4
# Skipped          : 0
=======================
```

Note that these all failed 4 times before being considered failures.

**Comment by sandreenko** [comment]
Failed tests for each run are different, so it is possible that it will be fixed with #13651. 
However, [the last job](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_ubuntu_tst_prtest/) failed with timeout and I will investigate that.

**Comment by sandreenko** [comment]
The timeout was an infrastructure problem that doesn't not repro.

I ran [a job ](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_ubuntu_tst_prtest/3)with 20 reruns, all tests passed after 8 reruns.

So I think we should wait until #13651 is fixed and then check this issue again.

**Comment by BruceForstall** [comment]
I'm ok temporarily bumping the max rerun count to 20, as the entire rerun infrastructure is hopefully temporary. If it would make jobs pass successfully, that would be good.

**Comment by sandreenko** [comment]
Release tests with fix from #17342 also fails with:
```
 Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: Arg_NullReferenceException
               
                  at System.Environment.FailFast(System.String)
```

This is clang 3.9 problem that we track in #16995.

**Comment by sandreenko** [comment]
I ran `JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b49809/b49809/b49809.sh` 200 times with clang 5.0 and there was no `Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: Arg_NullReferenceException` that we saw in [this run](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_ubuntu_tst_prtest/5/consoleText) with clang 3.9.

I think it is enough to say that clang 5.0 fixes this problem, however, it doesn't affect these failures:
`Segmentation fault      $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments` (tested on JIT/Performance/CodeQuality/Serialization/Serialize/Serialize.exe).

**Comment by RussKeldorph** [comment]
@janvorli It appears that #15074 or something related hits `Release` builds as well.  Do you think that tips the balance and we should go ahead and move to a newer Clang?  Can you suggest which version we should use?

**Comment by RussKeldorph** [comment]
@Petermarcu @ianhays

**Comment by sandreenko** [comment]
Before the docker update we had:
```
=======================
     Test Results
=======================
# CoreCLR Bin Dir  : 
# Tests Discovered : 11829
# Passed           : 10910
# Failed           : 114 (95, 108 in others runs)
# Skipped          : 805
=======================
```
[Now ](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_ubuntu_tst_prtest/6/consoleText)we have:
```
=======================
     Test Results
=======================
# CoreCLR Bin Dir  : 
# Tests Discovered : 11832
# Passed           : 10977
# Failed           : 50
# Skipped          : 805
=======================
```
and we now need 4 rerun to get clear run, but we still need them.

I am starting to investigate these failures one-by-one.

**Comment by sandreenko** [comment]
With cach flush fix [ubuntu arm32 Release job](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_release_ubuntu_tst_prtest/7/consoleText) passed without rerunning:
```
=======================
     Test Results
=======================
# CoreCLR Bin Dir  : 
# Tests Discovered : 11834
# Passed           : 11029
# Failed           : 0
# Skipped          : 805
=======================
59 minutes and 59 seconds taken to run CoreCLR tests.
```

**Comment by BruceForstall** [comment]
Nice! and all pri-1 under an hour!

**Comment by echesakovMSFT** [comment]
Closed in #17523
